### ISSUE TITLE
Is this still being maintained?

### ISSUE THREAD

**Issue by luisdavim** [jwilder/dockerize#120]
I see some lingering PRs is anyone still looking after this repo?

**Comment by luisdavim** [comment]
If it's not, maybe you could update the readme informing people to use https://github.com/presslabs/dockerize instead.

**Comment by carljmosca** [comment]
Good to know, thank you.

**Comment by jhult** [comment]
@jwilder, any updates on this?

**Comment by powerman** [comment]
I've just created new fork https://github.com/powerman/dockerize, designed as community replacement for original version until @jwilder will returns to work on original version.

* I've updated build to use Go 1.11 modules and automated binary releases using CI (automated docker releases are also available).
* I've merged @sychan fork and updated README to reflect changes (except `-euid` and `-egid` flags: there are existing tools for switching users, and it's much better to use these tools - they are more sophisticated and have well-known behaviour).
* I've merged @presslabs fork and updated README to reflect changes (cc: @calind). This adds some incompatibilities on template functions, but I've documented it and in general I think adding so many template functions worth it.

Version v0.7.0 was released with all these changes.

I'll be glad to add @sychan, @calind and other people as collaborators if they like to continue work using this fork (also I'm ready to convert repo from my personal to org and share admin privileges, if this will makes sense).

Please let me know if there are other forks or PRs with important changes which you like to see merged.

**Comment by powerman** [comment]
* I've merged feature-related changes from @moshloop fork and updated README to reflect changes.
* Features added in @danstiner, @dcrobbins, @bahlo forks are already included.
* Features added in @jmahowald fork are probably included too, but I'm not 100% sure.

Version v0.8.0 was released with all these changes.

**Comment by luisdavim** [comment]
That's great work @powerman, thanks.
Did you go through the currently open PRs in this repo?

**Comment by powerman** [comment]
> Did you go through the currently open PRs in this repo?

Yeah, I've checked both PRs and most of forks, but handled only ones which was ready to merge as is. I'm planning to handle rest of PRs and issues in this repo after I'll finish with project cleanup and refactoring - to make it easier to maintain it and make it testable it needs to be refactored first.

I've spend last days to [replace](https://github.com/powerman/dockerize/pull/8/files) `tail -f` implementation - old one had some race condition issue, few bugs and shortcomings which was really hard to fix in existing code base. I've checked other available implementations and was really surprised because they also have race condition issues and will lose lines under some conditions. So, I decide to implement reliable solution myself. This task looks deceptively trivial but in practice it's full of hidden issues. So it take more time than I expect to [implement](https://github.com/powerman/tail) it from scratch and [test](https://github.com/powerman/tail/blob/master/tail_test.go). To me it looks correct, but as this task has already proved it's not easy to do right I'll be happy if someone will review it, just in case.

I've tried to contact @jwilder at LinkedIn, but he didn't reply. Only communication channel left is https://twitter.com/jaswilder, but I didn't have Twitter account - maybe you'll be able to reach him there?

**Comment by powerman** [comment]
I've tried hard to refrain from making incompatible changes while this refactoring, but some changes was required to fix weird behaviour.
Let me know if you believe old behaviour was more useful/correct.

- flags: `-wait-http-header` now support file with header, to make it consistent with `-env-header`.
- flags: Improve help messages and use **more strict validation**.
- template: Fix chmod/chown generated files to match template files. (This may solve @xmlking [issue](https://github.com/xmlking/dockerize/commit/dcf101794e4519060cec604481afe8c66c81a3a5).)
- template: Fix rendering directory with templates to stdout.
- tail: Changed to **start from current file position** (work as `tail -n 0`).
- tail: Fixed to ensure it'll open existing files before running command.
- tail: Fixed to become more reliable and don't lose lines.
- tail: Logged messages mimics `tail` tool.
- command: **Add forwarding for SIGQUIT, SIGABRT, SIGUSR1, SIGUSR2, SIGALRM.**
- command: Add forwarding for SIGKILL (Linux only).
- command: **Don't send SIGKILL in 10s after forwarding HUP/INT/TERM.**
- command: Minimize possible conflict with dockerize exit code by using code **123 instead of 1** when dockerize itself has failed.
- Improve logging to make it more consistent.
- Huge refactoring to make code testable, simpler and easier to maintain.
- Add two hundred tests to check basic functionality.
- A lot of other minor cleanup and improvements.

Version [v0.9.0](https://github.com/powerman/dockerize/releases/tag/v0.9.0) was released with all these changes.

I consider code base stable now, no more large refactoring planned, so feel free to make changes you need. Next plans are:
- review issues and PR in this repo
- add more tests :smile: _(badge in README report coverage 7% but I believe it's actually about 80% - most current tests just run executable with different flags instead of calling functions, and coverage report ignores these tests)_
- add flag `-template-strict` to activate `Option("missingkey=error")` on templates
- consider adding flag `-exec` to replace dockerize process with given command (if there are `-stdout` or `-stderr` flags then run tail as child process) - this should let us run command as PID1 and avoid subtle issues related to signal forwarding etc.)

**Comment by jmahowald** [comment]
@powerman I've been using your template changes and really like them and the changes you made to the overall health of the code base, with tests, etc.  But I agree with @jwilder .  The change of the order of operations of several functions, even while documented can easily create confusion.  I think that most of the other non template functions in this project don't need a lot of love, but you've made a a very easy to use template library. dockerize-template maybe?
